src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -4536,11 +4548,24 @@ instruct maxD_reg(legRegD dst, legRegD a, legRegD b, legRegD tmp, legRegD atmp,
   %}
   ins_encode %{
     int vector_len = Assembler::AVX_128bit;
-    __ vblendvpd($btmp$$XMMRegister, $b$$XMMRegister, $a$$XMMRegister, $b$$XMMRegister, vector_len);
-    __ vblendvpd($atmp$$XMMRegister, $a$$XMMRegister, $b$$XMMRegister, $b$$XMMRegister, vector_len);
-    __ vmaxsd($tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister);
-    __ vcmppd($btmp$$XMMRegister, $atmp$$XMMRegister, $atmp$$XMMRegister, Assembler::_false, vector_len);
-    __ vblendvpd($dst$$XMMRegister, $tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister, vector_len);
+    XMMRegister mask = $b$$XMMRegister;
+    if (EnableX86ECoreOpts) {
+      __ vpxor($tmp$$XMMRegister, $tmp$$XMMRegister, $tmp$$XMMRegister, vector_len);
+      __ vpcmpgtq($tmp$$XMMRegister, $tmp$$XMMRegister, mask, vector_len);
+      mask = $tmp$$XMMRegister;
+    }
+    __ vblendvpd($btmp$$XMMRegister, $b$$XMMRegister, $a$$XMMRegister, mask, vector_len, true, $atmp$$XMMRegister);
+    __ vblendvpd($atmp$$XMMRegister, $a$$XMMRegister, $b$$XMMRegister, mask, vector_len, true, $tmp$$XMMRegister);
+
+    if ($dst$$XMMRegister == $btmp$$XMMRegister) {
+      __ vmaxsd($btmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister);
+      __ vcmppd($tmp$$XMMRegister, $atmp$$XMMRegister, $atmp$$XMMRegister, Assembler::_false, vector_len);
+      __ vblendvpd($dst$$XMMRegister, $btmp$$XMMRegister, $atmp$$XMMRegister, $tmp$$XMMRegister, vector_len, true, $tmp$$XMMRegister);
+    } else {
+      __ vmaxsd($tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister);
+      __ vcmppd($btmp$$XMMRegister, $atmp$$XMMRegister, $atmp$$XMMRegister, Assembler::_false, vector_len);
+      __ vblendvpd($dst$$XMMRegister, $tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister, vector_len, true, $btmp$$XMMRegister);
+    }
```

Comment:
Please move to a new macro assembly routine.

Reply:
moved to existing routine